target/cheri-common/op_helper_cheri_common.c

```diff
@@ -690,37 +690,45 @@ DEFINE_CHERI_STAT(csetaddr);
 DEFINE_CHERI_STAT(candaddr);
 DEFINE_CHERI_STAT(cfromptr);
 
-static void cincoffset_impl(CPUArchState *env, uint32_t cd, uint32_t cb,
-                            target_ulong rt, uintptr_t retpc,
-                            struct oob_stats_info *oob_info)
+static inline QEMU_ALWAYS_INLINE void
+cincoffset_impl(CPUArchState *env, uint32_t cd, uint32_t cb, target_ulong rt,
+                uintptr_t retpc, struct oob_stats_info *oob_info)
```

Comment:
That `#else`/`#endif` mid-way through this definition feels ugly.  Since you're in the neighborhood, maybe do away with it and just `#ifdef`-guard the `->num_uses++`?

Reply:
Yeah, `struct oob_stats_info *oob_info` will implicitly declare the struct, it's only an error if you attempt to dereference the incomplete type.

Reply:
Having said that, that's not new in this code :)

Reply:
Sounds like a good suggestions, I'll do that as a follow-up cleanup.